idea.js

```diff
@@ -1,23 +1,22 @@
 class Idea {
-  constructor(id, title, body, starred) {
+  constructor(id, title, body) {
     this.id = id || Date.now();
     this.title = title;
     this.body = body;
-    this.starred = starred || false;
+    this.starred = false;
   }
 
   saveToStorage() {
     var stringifiedIdeas = JSON.stringify(ideaArray);
     localStorage.setItem("ideaLocalStorage", stringifiedIdeas);
   }
 
-  updateIdea(){
-    this.starred = !this.starred;
```

Comment:
I'm not sure I follow why you took this out. 

Obviously, this is still creating the desired result on the DOM and in `localStorage`, but you aren't actually changing the `starred` value of this particular `Idea`. 

With this implementation, it will always be set to false. What's the point of having the property on the class if it's only used when you're creating the initial idea? 🤷‍♂

Reply:
When creating a card there should be no other option than false, so it isn't needed as a parameter. There isn't a way to favorite a card at time of creation in the comp. Maybe I'm misunderstanding your question...

Reply:
So, the `ideasArray` variable that you have in `main.js` is holding onto all of those `Idea`s, right? The way this is set up (eg by removing `this.starred = !this.starred`) means that you're never really updating the actual `Idea`, only the DOM and local storage.

Reply:
Ideally, all three should match. If you want to see what I'm talking about, go into your code and `console.log(ideaArray)` each time you update a favorite. You should see that the actual `Idea`s that you're holding onto aren't having their `starred` value updated, even though you're updating it on the DOM and in local storage.

I'll check it out as well, just to make sure I know what I'm talking about 🙂

Reply:
I just checked and I was right, the classes aren't updating when you click a card and mark it as a favorite. Consider it this way: When you pull a card from local storage and it's already favorited, why would you want the `starred` prop to default to false?

Your current order of operations is basically this:
1. Pull a favorited card from local storage
2. Create a new `Idea` with those values (even though the card has `starred: true`, it is being set to `false` with the current setup)
3. Render the card to the DOM (by default it is not favorited due to the default of `false`)
4. Now your `setFavoriteStyle` runs through local storage _again_ and marks all of the favorites

This approach basically runs through local storage twice, when you can do all of this by only going through once. To accomplish this, your order of operations should probably look more like this:
1. Pull a favorited card from local storage
2. Create a new `Idea` with all values (including `starred`)
3. When you render the card to the DOM it _alread knows_ that it's favorited, so you can add the appropriate class

Reply:
Ok, I see what you mean now. Added that back in.